suite-native/passphrase/src/PassphraseFormModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useRef } from 'react';
+import { Modal, TextInput } from 'react-native';
+
+import { useSetAtom } from 'jotai';
+
+import { Form, TextInputField, useForm } from '@suite-native/forms';
+import { yup } from '@trezor/validation';
+import { Box, Button, VStack } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { useTranslate } from '@suite-native/intl';
+
+import { isPassphraseModalVisibleAtom } from './passphraseModalVisibilityAtom';
+
+const PASSPHRASE_MIN_LENGTH = 1;
+const PASSPHRASE_MAX_LENGTH = 50;
+
+const passphraseFormSchema = yup.object({
+    passphrase: yup
+        .string()
+        .required('Empty passphrase.')
+        .max(PASSPHRASE_MAX_LENGTH)
+        .min(PASSPHRASE_MIN_LENGTH),
+});
+
+type PassphraseFormValues = yup.InferType<typeof passphraseFormSchema>;
+
+const modalBackgroundOverlayStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    backgroundColor: utils.transparentize(0.3, utils.colors.backgroundNeutralBold),
+}));
+
+const formStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.backgroundSurfaceElevation1,
+    borderBottomLeftRadius: utils.borders.radii.large,
+    borderBottomRightRadius: utils.borders.radii.large,
+}));
+
+export const PassphraseFormModal = () => {
+    const { translate } = useTranslate();
+
+    const passphraseInputRef = useRef<TextInput>(null);
+
+    const setPassphraseModalVisibility = useSetAtom(isPassphraseModalVisibleAtom);
+
+    const { applyStyle } = useNativeStyles();
+
+    const form = useForm<PassphraseFormValues>({
+        validation: passphraseFormSchema,
+        defaultValues: {
+            passphrase: '',
+        },
+    });
+
+    const { handleSubmit } = form;
+
+    useEffect(() => {
+        passphraseInputRef.current?.focus();
+    }, [passphraseInputRef]);
+
+    const handleCreateHiddenWallet = handleSubmit(() => {
+        setPassphraseModalVisibility(false);
+        // TODO create wallet
+    });
+
+    return (
+        <Modal transparent>
+            <Box style={applyStyle(modalBackgroundOverlayStyle)}>
+                <Form form={form}>
+                    <VStack spacing="medium" padding="medium" style={applyStyle(formStyle)}>
+                        <TextInputField
```

Comment:
OK for now, but we would need to think about the input type for passhprase. On desktop it is custom made password type (input is masked as dots, but it's custom made so password managers are not suggesting you to fill in/remember).

<img width="542" alt="image" src="https://github.com/trezor/trezor-suite/assets/3729633/1dc17868-784b-400d-8edc-f0d1d83e1f51">

Reply:
Very good catch! I will create an issue for masking the input. Didn't realise this.

Reply:
Issue created in epic